### ISSUE TITLE
[Request] Subcategory

### ISSUE THREAD

**Issue by mrdroops** [inorichi/tachiyomi#1803]
**App version:**
0.8.0

**Issue/Request:**
    I would like to make a suggestion: to be able to add subcategories. Ex: when I create a category "Mangas", could have as add subcategories as the genres of manga.

**Comment by as280093** [comment]
@mrdroops rather then that in library filter add all the genre as in the extension

it could be better

**Comment by mrdroops** [comment]
Of course, so too, but as for example in my case, I am Brazilian, and I have saved in my manga library in Portuguese and English, however, had to be created many categories, would be more organized. At least I think.

**Comment by Schnoinsch** [comment]
The possible use of subcategories are big.I kinda think of it as a mandatory thing, to get big librarys in order. Not sure since i never wrote a single line of code but it seems not to hard to implement such a feature.

**Comment by arkon** [comment]
From #3750:

https://user-images.githubusercontent.com/70755549/92233897-baa0de80-eece-11ea-94a0-e0e8a6826cc2.jpg

**Comment by monMusu** [comment]
Hi made a detailed mockup for the Virtual Folders.
![V-folders](https://user-images.githubusercontent.com/70755549/99935117-efae2600-2d85-11eb-97b2-c0721fc6b993.png)

**Comment by monMusu** [comment]
I updated the "Virtual Folders" mock up with a detailed layout for creating the folders and subfolders.
![V-folders](https://user-images.githubusercontent.com/70755549/99936080-44eb3700-2d88-11eb-93cf-428087064b40.png)

**Comment by kiobu** [comment]
Any update on this? Was about to suggest the same thing before finding this issue.

**Comment by GlassedSilver** [comment]
I'd like to point out that this should allow for more than one hierarchy.

I'd like to sort by categories of various kinds and then within those I'd love to be able to group franchises, so sequels, spin-offs,  etc all sit within one "franchise" level of hierarchy.

**Comment by soramui** [comment]
Is this feature currently being actively worked on?

I'd suggest to completely replace the top slider for category with folders that can be accessed the same way as regular manga and can be put inside another folder, just like in a regular file manager. This way there is no need to maintain different code for category and subcategory and navigating the app would also be more intuitive for people who are already used to UI in windows/mac/linux/etc.

**Comment by FourTOne5** [comment]
> Is this feature currently being actively worked on?
> 
It's hardly being worked on
> I'd suggest to completely replace the top slider for category with folders that can be accessed the same way as regular manga and can be put inside another folder, just like in a regular file manager. This way there is no need to maintain different code for category and subcategory and navigating the app would also be more intuitive for people who are already used to UI in windows/mac/linux/etc.
>
Why would someone want a file manager like UI in a manga reader app.

**Comment by soramui** [comment]
Because it's not fun scrolling between categories in its current form when you have 50+ categories (and yes, i do have that many for some reason), just clicking on a folder instead would be better

**Comment by GlassedSilver** [comment]
> > Is this feature currently being actively worked on?
> 
> It's hardly being worked on
> 
> > I'd suggest to completely replace the top slider for category with folders that can be accessed the same way as regular manga and can be put inside another folder, just like in a regular file manager. This way there is no need to maintain different code for category and subcategory and navigating the app would also be more intuitive for people who are already used to UI in windows/mac/linux/etc.
> 
> Why would someone want a file manager like UI in a manga reader app.

I think it's important to make a distinction here between a file-manager-like UI and file-manager-style flexibility.

Also, I don't really think of these as folders like in a "dumb" file manager. Rather there is a distinction (hopefully) between categories, grouped series (a franchise) and possible sub-categories.

None of these should be required to be used I guess, not everyone needs that much order, e.g. people who just bring around 5-10 manga.

But if you intend to fill an entire SD card with manga because you dedicate an entire Android tablet to this, then you will very much need and want good grouping and discoverability.

**Comment by kiobu** [comment]
This is such an incredibly useful and obvious feature that I'm floored it hasn't received any attention yet honestly.

**Comment by saud-97** [comment]
> I'd suggest to compl etely replace the top slider for category with folders that can be accessed the same way as regular manga and can be put inside another folder, just like in a regular file manager. This way there is no need to maintain different code for category and subcategory and navigating the app would also be more intuitive for people who are already used to UI in windows/mac/linux/etc.

I think what you mean is something like a breadcrumb.

![image](https://user-images.githubusercontent.com/39028181/168231872-86d790f3-4881-482e-85ea-cdbc49d7d538.png)

And I agree that some people might use this feature to structure their Library and organize it.

**Comment by GlassedSilver** [comment]
This would also greatly help this endevour: https://github.com/hkalexling/Mango/issues/33#issuecomment-916833727
